cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -65,6 +69,10 @@ type InitConfiguration struct {
 type ClusterConfiguration struct {
 	metav1.TypeMeta
 
+	// Metadata is the standard object metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	Metadata *metav1.ObjectMeta
```

Comment:
i went with not embedding the structure due to the conflicting ClusterConfiguration.ClusterName and ObjectMeta.ClusterName.

i don't think ObjectMeta.ClusterName is being used in core currently.

the old PR and long discussion was here:
https://github.com/kubernetes/kubernetes/pull/84357

Reply:
Note that embedding ObjectMeta into CRDs usually has implications for validation. IIRC, we removed ObjectMeta from our types and used only the fields needed because ObjectMeta is validated in the api server separately, and the object fails validation due to the custom logic that's in there.

Why do we need this here though?